components/global_onboarding_list/onboarding_checklist.tsx

```diff
@@ -0,0 +1,269 @@
+            name: currentUserId,
+            value: 'true',
+        }];
+        dispatch(savePreferences(currentUserId, preferences));
+        trackEvent('ui', 'onboarding_checklist_dismissed');
+    }, [currentUserId]);
+
+    const taskLabels = [
+        <FormattedMessage
+            key={1}
+            id='onboarding_checklist.task_channels_tour'
+            defaultMessage='Take a tour of channels'
+        />,
+        <FormattedMessage
+            key={2}
+            id='onboarding_checklist.task_boards_tour'
+            defaultMessage='Manage tasks with your first board'
+        />,
+        <FormattedMessage
+            key={3}
+            id='onboarding_checklist.task_invite'
+            defaultMessage='Invite team members to the workspace'
+        />,
+        <FormattedMessage
+            key={4}
+            id='onboarding_checklist.task_complete_profile'
+            defaultMessage='Complete your profile'
+        />,
+        <FormattedMessage
+            key={5}
+            id='onboarding_checklist.task_download_apps'
+            defaultMessage='Download the Desktop and Mobile Apps'
+        />,
+        <FormattedMessage
+            key={6}
+            id='onboarding_checklist.task_system_console'
+            defaultMessage='Visit the System Console to configure your worksace'
+        />,
+    ];
+    const closeTaskList = useCallback(() => {
+        setOpen(false);
+    }, []);
+
+    return (
+        <>
+            <CompletedAnimation completed={completedCount === taskLabels.length}/>
+            <Button
+                onClick={() => setOpen(!open)}
+                ref={trigger as unknown as React.RefObject<HTMLButtonElement>}
+                open={open}
+            >
+                <Icon glyph={open ? 'close' : 'playlist-check'}/>
+                <span>{taskLabels.length - completedCount}</span>
+            </Button>
+            <TaskListPopover
+                isVisible={open}
+                trigger={trigger}
+                onClick={closeTaskList}
+            >
+                <TaskItems className={open ? 'open' : ''}>
+                    {completedCount === taskLabels.length ? <Completed dismissAction={dismissChecklist}/> : (
+                        <>
+                            <h1>
+                                <FormattedMessage
+                                    id='next_steps_view.welcomeToMattermost'
+                                    defaultMessage='Welcome to Mattermost'
+                                />
+                            </h1>
+                            <p>
+                                <FormattedMessage
+                                    id='onboarding_checklist.main_subtitle'
+                                    defaultMessage="Let's get up and running."
+                                />
+                            </p>
+                            <img
+                                src={checklistImg}
```

Comment:
I think we should need to add to wrap inside a div with some color as placeholder, As image will take sometime to load and that’s creating issue.